tests/unit_tests/test_primary_module.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+from unittest.mock import Mock, MagicMock
+from resolve.resolver import Resolver
+from modules.primary_monitoring.module import PrimaryMonitoringModule
+from resolve.enums import Function, Module
+from modules.enums import PrimaryMonitoringEnums as enums
+from modules.constants import V_STATUS, PRIM, NEED_CHANGE, NEED_CHG_SET
+
+
+class TestPredicatesAndAction(unittest.TestCase):
+
+    def setUp(self):
+        self.resolver = Resolver()
+
+    # Macros
+    def test_clean_state(self):
+        # Let primary be 0
+        self.resolver.execute = MagicMock(return_value = 0)
+        primary_mod = PrimaryMonitoringModule(0, self.resolver, 6, 1)
+        # Create a non-default vcm
+        primary_mod.vcm = [{V_STATUS: enums.V_CHANGE, PRIM: 1, NEED_CHANGE: True, NEED_CHG_SET: {0,2}} for i in range(6)]
+        # Check so that vcm is not in default state
+        self.assertNotEqual(primary_mod.vcm,
+            [{V_STATUS: enums.OK, PRIM: 0, NEED_CHANGE: False, NEED_CHG_SET: set()} for i in range(6)])
+        # Clean the vcm and check that all elements are set to the default
+        primary_mod.clean_state()
+        self.assertEqual(primary_mod.vcm,
+            [{V_STATUS: enums.OK, PRIM: 0, NEED_CHANGE: False, NEED_CHG_SET: set()} for i in range(6)])
+
+    def test_sup_change(self):
+        # Let primary be 0
+        self.resolver.execute = MagicMock(return_value = 0)
+        primary_mod = PrimaryMonitoringModule(0, self.resolver, 6, 1)
+
+        # All nodes are needing change
+        primary_mod.vcm = [{V_STATUS: enums.OK, PRIM: 0, NEED_CHANGE: True, NEED_CHG_SET: {0,1,2,3,4,5}} for i in range(6)]
+        self.assertTrue(primary_mod.sup_change(6))
+        # Length of the processor set will not be 7
+        self.assertFalse(primary_mod.sup_change(7))
+
+        # The intersection will not be large enough, all have same prim
+        # but only 0,1,2 is in NEED_CHG_SET
+        primary_mod.vcm = [{V_STATUS: enums.OK,
```

Comment:
Try the other case as well, the amount of processors that have `NEED_CHG_SET = True` are enough and all agree on the same primary